dev-docs/RFCs/v8.1/gpu-accelerated-geo-fencing.md

```diff
@@ -0,0 +1,153 @@
+
+### PolygonTexture
+
+Axis aligned bounding box of polygons is first computed. A Framebuffer is created that has same aspect reatio as bounding box and it is cleared with a custom clear color (can be black). User provided polygons are triangulated and their positions are normalized to bounding box and are rendered to above offline Framebuffer using a custom color (we can just use one of RGB channels).
+
+### Filtering
+
+When rendering user provided data, for each data element, its positions is transformed into same space that above texture is generated and it position (XY) is normalized to a [0, 1] range and then we first perform trivial check agains bounding box (for range [0, 1]) to clip agains bounding box and then perform texture sample from above texture. The result gives whether the object is in the polygon or not.
+
+## Problem Classifications and proposed API
+
+### Filtering points
+
+We can perform a point in polygon test of each object and discard the rendering if test fails. One are more layer can be marked to generate the filter texture and one or more layers can be marked to use this texture as a mask for filtering. In layer's vertex shader, a texture sample is performed to determine if the object is inside or outside.
+
+Entire process of filtering can be divided into two steps, 1. Generation of filter texture and 2. performing actual filtering.
+
+* API : Following new components will be added
+
+#### GeoFilterEffect
+
+This effect generates the required filter texture. Internally it creates a new render pass to render data in world space to a offline texture.
+
+During pre render pass:
+- All layers are processed, and if a layer has 'geoFilterMask' prop set, a bounding box of its data is calculated (optionally can be provided by the layer).
+- All above bounding boxes are merged at each step and used to build the texture mask.
+- This effect class is responsible for providing following uniforms that will be consumed by following shader module 'TextureFilterModule'
+
+1. `textureTransform` (Array, [xTranslation, yTranslation, xScale, yScale]) : used to transform each data point into required clipping space.
+2. `maskTexture` (Texture2D) : texture that is sampled to determine if a given point is inside or outside.
+
+#### TextureFilterModule
+
+A new shader module is added to perform sampling from the texture and filter the points, defines following shader functions:
+
+- `geoFilter_setValue` : Takes, object's world positions, transforms it to mask texture space, performs sampling and sets `geoFilter_value` to either 0 (if outside the defined region) or to and id ( >= 1).
```

Comment:
`value` is not very descriptive. Call the function `geoFilter_setPosition` and the varying `geoFilter_isVisible` for clarity?